### PR TITLE
[Codegen 77] Extract the functions to compute partial properties in the Flow and TypeScript parsers.

### PR DESCRIPTION
**Pull request by MaeIg**
## Summary

This PR aims to extract  the switch(configType) block from the buildSchema function into a separate function in a shared file between typescript and flow. It is a task of #34872:
> [Codegen 77 - assigned to @MaeIg] Extract the functions to compute partial properties from the index.js file ([Flow](https://github.com/facebook/react-native/blob/main/packages/react-native-codegen/src/parsers/flow/modules/index.js#L180-L194) and [TypeScript](https://github.com/facebook/react-native/blob/main/packages/react-native-codegen/src/parsers/typescript/modules/index.js#L261-L278))in the Flow and TypeScript parsers.

## Changelog

[Internal] [Changed] - Extract the functions to compute partial properties in the Flow and TypeScript parsers.

## Test Plan

yarn flow:
image

yarn lint:
image

yarn test
image

### CODE REVIEW

**Comment by rshest** [comment]
@MaeIg Could you please add tests for that as well?

**Comment by MaeIg** [comment]
> @MaeIg Could you please add tests for that as well?

Hello @rshest,

I added a test for each parser. I'm not sure these tests are useful because there is only one branch and everything is already covered by higher level tests.

File: packages/react-native-codegen/src/parsers/parser.js [hunk 1]
```diff
@@ -181,4 +183,25 @@ export interface Parser {
...
+
+  /**
+   * Given an array of properties from a Partial type, it returns an array of remaped properties.
+   * @paramater properties: properties from a Partial types.
+   * @parameter hasteModuleName: a string with the native module name.
+   * @paramater types: a map of type declarations.
+   * @paramater aliasMap: a map of type aliases.
+   * @paramater enumMap: a map of type enums.
+   * @paramater tryParse: a parser error capturer.
+   * @paramater cxxOnly: a boolean specifying if the module is Cxx only.
```
Comment (rshest) [inline packages/react-native-codegen/src/parsers/parser.js:195]: "parameter"?.. ;)

Comment (MaeIg) [reply packages/react-native-codegen/src/parsers/parser.js:195]: Ups sorry for the typo!

I see that you fixed the typo and merged the PR, thank you! :)

**Pull request closed without merging**